cis194/week7/zehua/test/JoinListBufferSpec.hs

```diff
@@ -0,0 +1,95 @@
+module JoinListBufferSpec where
+
+import           Buffer
+import           Control.Monad   (mapM_)
+import           Data.List.Index (setAt)
+import           JoinList
+import           JoinListBuffer
+import           Scrabble
+import           Sized
+import           StringBuffer
+import           Test.Hspec
+
+-- create a Single score size annotated JoinList
+sss :: Int -> Int -> String -> JoinList (Score, Size) String
+sss s1 s2 = Single ((Score s1), (Size s2))
+
+-- create an Append score size annotated JoinList
+ass :: Int -> Int
+           -> JoinList (Score, Size) String
+           -> JoinList (Score, Size) String
+           -> JoinList (Score, Size) String
+ass s1 s2 = Append ((Score s1), (Size s2))
+
+dataEmptyJL :: JoinList (Score, Size) String
+dataEmptyJL = Empty
+
+dataA2QList :: [String]
+dataA2QList = ["abc", "def", "hij", "klm", "opq"]
+
+dataA2QString :: String
+dataA2QString = unlines dataA2QList
+
+dataA2QJL :: JoinList (Score, Size) String
+dataA2QJL = ass 50 5
+            (ass 27 3
+              (ass 14 2
+                (sss 7 1 "abc")
+                (sss 7 1 "def"))
+              (sss 13 1 "hij"))
+            (ass 23 2
+              (sss 9 1 "klm")
+              (sss 14 1 "opq"))
+
+infix 1 `shouldBeJLEqualTo`
+shouldBeJLEqualTo :: JoinList (Score, Size) String -> JoinList (Score, Size) String -> Expectation
+actual `shouldBeJLEqualTo` expected = do
+  toLines actual `shouldBe` toLines expected
+  getJoinListScore actual `shouldBe` getJoinListScore expected
+  getJoinListSize actual `shouldBe` getJoinListSize expected
```

Comment:
not sure if this is the right way to build helper assert functions in Haskell

Reply:
they are just normal functions so this is fine.